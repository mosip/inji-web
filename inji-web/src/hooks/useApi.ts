import {useState} from "react";
import {ApiRequest} from "../types/data";
import {api, MethodType} from "../utils/api";

export interface NetworkResult<T> {
    data: T | null;
    error: Error | null;
    status: number | null;
    loading: boolean;
    headers: object;
}

interface RequestConfig {
    url?: string;
    headers?: Record<string, string>;
    body?: any;
    apiRequest: ApiRequest;
}

interface UseApiReturn<T> {
    data: T | null;
    error: Error | null;
    loading: boolean;
    status: number | null;
    fetchData: (arg0: RequestConfig) => Promise<NetworkResult<T>>;
}

export function useApi<T = any>(): UseApiReturn<T> {
    const [data, setData] = useState<T | null>(null);
    const [error, setError] = useState<Error | null>(null);
    const [loading, setLoading] = useState<boolean>(false);
    const [status, setStatus] = useState<number | null>(null);

    async function fetchData({
                                 headers,
                                 body,
                                 apiRequest,
                                 url = undefined
                             }: RequestConfig): Promise<NetworkResult<T>> {
        setLoading(true);
        setError(null);
        setStatus(null);

        let result: NetworkResult<T> = {
            data: null,
            error: null,
            status: null,
            loading: true,
            headers: {}
        };

        try {
            const response = await api.instance.request({
                url: url ?? apiRequest.url(),
                method: MethodType[apiRequest.methodType],
                headers,
                data: body,
                withCredentials: apiRequest.credentials === "include",
                responseType: apiRequest.responseType ?? "json",
            });

            console.log("API response:", response);
            setData(response.data);
            setStatus(response.status);

            result = {
                data: response.data,
                error: null,
                status: response.status,
                loading: false,
                headers: response.headers || {}
            };
        } catch (err: any) {
            const parsedError = err instanceof Error ? err : new Error("Unknown error");

            setError(parsedError);
            setStatus(err?.response?.status || null);

            result = {
                data: null,
                error: parsedError,
                status: err?.response?.status || null,
                loading: false,
                headers: err?.response?.headers || {}
            };
        } finally {
            setLoading(false);
        }

        return result;
    }

    return {data, error, loading, status, fetchData};
}
