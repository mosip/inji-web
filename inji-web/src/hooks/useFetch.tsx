import {useState} from "react";
import {api, MethodType} from "../utils/api";

export enum RequestStatus {
    LOADING,
    DONE,
    ERROR
}

export const useFetch = () => {
    const [state, setState] = useState<RequestStatus>(RequestStatus.LOADING);
    const [error, setError] = useState<string>("");

    const fetchRequest = async (uri: string, method: MethodType, header: any, body?: any) => {
        try {
            setState(RequestStatus.LOADING);
            const response = await fetch(`${uri}`, {
                method: MethodType[method],
                headers: header,
                body: body,
            });

            if (uri.indexOf("download") !== -1) {
                setState(RequestStatus.DONE);
                return await response.blob();
            }
            if (!response.ok) {
                setState(RequestStatus.ERROR);
                return response;
            }
            setState(RequestStatus.DONE);
            return await response.json();
        } catch (e) {
            setState(RequestStatus.ERROR);
            setError("Error Happened");
        }
    };
    return {state, error, fetchRequest};
}



