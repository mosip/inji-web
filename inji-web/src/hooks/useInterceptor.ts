import {useEffect} from 'react';
import {useNavigate, useLocation} from 'react-router-dom';
import {api} from "../utils/api";

export function useAxiosInterceptor() {
    const navigate = useNavigate();
    const location = useLocation();

    const interceptor = api.instance.interceptors.response.use(function (response: any) {
        console.log("interceptor response", response);
        return response;
    }, function (error: { response: { status: number; }; }) {
        console.log("interceptor error out", error.response);
        if( error.response && error.response.status === 401) {
            navigate("user/passcode", {
                state: {
                    from: location.pathname + location.search + location.hash
                }
            })
        }
        return Promise.reject(error);
    });

    useEffect(() => {
        return () => {
            api.instance.interceptors.response.eject(interceptor);
        };
    }, [navigate, location, interceptor]);
}
