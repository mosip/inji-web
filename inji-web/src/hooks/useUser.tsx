import {useState, useEffect} from "react";
import { api } from "../utils/api";

export type User = {
    displayName: string;
    profilePictureUrl: string;
}

type ErrorObj = {
    errorCode: string;
    errorMessage: string;
}

export function useUser() {
    const [user, setUser] = useState<User | null>(null);
    const [errorObj, setErrorObj] = useState<ErrorObj | null>(null);

    useEffect(() => {
        const userData = localStorage.getItem("user");
        if (userData) {
            try {
                setUser(JSON.parse(userData));
            } catch (e) {
                console.error("Invalid user data in localStorage");
                localStorage.removeItem("user");
            }
        }
    }, []);

    const saveUser = (userData: User) => {
        localStorage.setItem("user", JSON.stringify(userData));
        setUser(userData);
    };

    const removeUser = () => {
        localStorage.removeItem("user");
        setUser(null);
    };

    const fetchUserProfile = async () => {
        try {
            const response = await fetch(api.fetchUserProfile.url(), {
                method: api.fetchUserProfile.methodType === 0 ? "GET" : "POST",
                headers: {...api.fetchUserProfile.headers()},
                credentials: "include"
            });
    
            const responseData = await response.json();
    
            if (!response.ok) {
                throw responseData;
            }
    
            const userData: User = {
                "displayName": responseData.display_name,
                "profilePictureUrl": responseData.profile_picture_url
            };

            saveUser(userData);
        } catch (error) {
            console.error("Error occurred while fetching user profile:", error);
            setUser(null);
            setErrorObj(error as ErrorObj)
            throw error;
        }
    };

    return {user, errorObj, saveUser, removeUser, fetchUserProfile};
}
