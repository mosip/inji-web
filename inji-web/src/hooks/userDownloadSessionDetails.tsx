import {useEffect, useState} from "react";
import {RequestStatus} from "./useFetch";

interface SessionStatus {
    downloadStatus: RequestStatus;
}

interface SessionsMap {
    [credentialType: string]: SessionStatus;
}

const getSessionsFromLocalStorage = (): SessionsMap => {
    try {
        const sessions = localStorage.getItem('downloadSessions');
        return sessions ? JSON.parse(sessions) : {};
    } catch (e) {
        console.error("Error parsing download sessions from localStorage:", e);
        return {};
    }
};

const dispatchLocalStorageChangeEvent = () => {
    window.dispatchEvent(new CustomEvent(LOCAL_STORAGE_CHANGE_EVENT));
};

export const LOCAL_STORAGE_CHANGE_EVENT = 'downloadSessionsChange';

export const useDownloadSessionDetails = () => {
    const [downloadSessions, setDownloadSessions] = useState<SessionsMap>(getSessionsFromLocalStorage());

    useEffect(() => {
        const updateSessionsFromEvent = () => {
            setDownloadSessions(getSessionsFromLocalStorage());
        };

        window.addEventListener(LOCAL_STORAGE_CHANGE_EVENT, updateSessionsFromEvent);

        const handleStorageChange = (event: StorageEvent) => {
            if (event.key === 'downloadSessions') {
                updateSessionsFromEvent();
            }
        };
        window.addEventListener('storage', handleStorageChange);

        return () => {
            window.removeEventListener(LOCAL_STORAGE_CHANGE_EVENT, updateSessionsFromEvent);
            window.removeEventListener('storage', handleStorageChange);
        };
    }, []);

    const addOrUpdateSession = (credentialType: string, downloadStatus: RequestStatus) => {
        const sessions = getSessionsFromLocalStorage();
        const updatedSessions = {...sessions, [credentialType]: {downloadStatus}}
        localStorage.setItem('downloadSessions', JSON.stringify(updatedSessions));
        dispatchLocalStorageChangeEvent();
    };

    const removeActiveSession = (credentialType: string) => {
        const sessions = getSessionsFromLocalStorage();
        const {[credentialType]: _, ...updatedSessions} = sessions;
        localStorage.setItem('downloadedSessions', JSON.stringify(updatedSessions));
        dispatchLocalStorageChangeEvent();
    };

    return {
        addOrUpdateSession,
        removeActiveSession,
        downloadSessions
    };
};