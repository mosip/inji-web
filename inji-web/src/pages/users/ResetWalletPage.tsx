import React from 'react';
import {SolidButton} from '../../components/Common/Buttons/SolidButton';
import {Trans, useTranslation} from 'react-i18next';
import {useUser} from '../../hooks/useUser';
import {api} from '../../utils/api';
import {useCookies} from 'react-cookie';
import {useLocation, useNavigate} from 'react-router-dom';
import {toast} from 'react-toastify';
import {ResetWalletPageStyles} from '../../styles/pages/ResetWalletPageStyles';
import {BackgroundDecorator} from '../../components/Common/BackgroundDecorator';
import {BackArrowButton} from '../../components/Common/Buttons/BackArrowButton';
import {InfoIcon} from '../../components/Common/Icons/InfoIcon';
import {ROUTES} from "../../constants/Routes";

interface InstructionItem {
    id: string;
    className: string;
    content: React.ReactNode;
}

interface InstructionContentProps {
    instructionItems: InstructionItem[];
    className: string;
    testId: string;
}

const InstructionContent: React.FC<InstructionContentProps> = ({ 
    instructionItems, 
    className, 
    testId 
}) => {
    return (
        <div data-testid={testId} className={className}>
            {instructionItems.map((item) => (
                <p
                    key={item.id}
                    data-testid={item.id}
                    className={item.className}
                >
                    {item.content}
                </p>
            ))}
        </div>
    );
};

export const ResetWalletPage: React.FC = () => {
    const {removeWallet, walletId} = useUser();
    const {t} = useTranslation('ResetWalletPage');
    const [cookies] = useCookies(['XSRF-TOKEN']);
    const navigate = useNavigate();
    const location = useLocation();

    const handleBackNavigation = () => {
        navigate(ROUTES.PASSCODE);
    };

    const handleForgotPasscode = async () => {
        try {
            const response = await fetch(
                api.deleteWallet.url(location.state?.walletId ?? walletId),
                {
                    method: 'DELETE',
                    headers: {
                        ...api.deleteWallet.headers(),
                        'X-XSRF-TOKEN': cookies['XSRF-TOKEN']
                    },
                    credentials: 'include'
                }
            );
            if (!response.ok) {
                throw await response.json();
            }
            removeWallet();
            navigate(ROUTES.PASSCODE);
        } catch (error) {
            console.error('Error occurred while deleting Wallet:', error);
            toast.error(t('resetFailure'));
        }
    };
    
    const instructionItems: InstructionItem[] = [
        {
            id: "text-reset-question",
            className: ResetWalletPageStyles.instructionQuestion,
            content: t('resetInstruction.question')
        },
        {
            id: "text-reset-info1",
            className: ResetWalletPageStyles.instructionText,
            content: t('resetInstruction.info1')
        },
        {
            id: "text-reset-info2",
            className: ResetWalletPageStyles.instructionText,
            content: (
                <Trans
                    i18nKey={t('resetInstruction.info2.message')}
                    ns="ResetWalletPage"
                    values={{
                        highlighter1: t('resetInstruction.info2.highlighter1', { ns: 'ResetWalletPage' }),
                        highlighter2: t('resetInstruction.info2.highlighter2', { ns: 'ResetWalletPage' })
                    }}
                    components={{
                        strong: <strong className={ResetWalletPageStyles.instructionTextStrong} />
                    }}
                />
            )
        },
        {
            id: "text-reset-info3",
            className: ResetWalletPageStyles.instructionText,
            content: (
                <Trans
                    i18nKey={t('resetInstruction.info3.message')}
                    ns="ResetWalletPage"
                    values={{
                        highlighter: t('resetInstruction.info3.highlighter', { ns: 'ResetWalletPage' })
                    }}
                    components={{
                        strong: <strong className={ResetWalletPageStyles.instructionTextStrong} />
                    }}
                />
            )
        }
    ];

    return (
        <div
            className={ResetWalletPageStyles.backdrop}
            data-testid="backdrop-reset-wallet"
        >
            <div className={ResetWalletPageStyles.container}>
                <BackgroundDecorator
                    logoSrc={require('../../assets/Logomark.png')}
                    logoAlt="Inji Web Logo"
                    logoTestId="logo-inji-web"
                />
                <div className={ResetWalletPageStyles.contentWrapper}>
                    <div className={ResetWalletPageStyles.header}>
                        <h1
                            className={ResetWalletPageStyles.title}
                            data-testid="title-reset-wallet"
                        >
                            {t('title')}
                        </h1>
                        <div className={ResetWalletPageStyles.subHeader}>
                            <BackArrowButton
                                onClick={handleBackNavigation}
                                btnClassName={
                                    ResetWalletPageStyles.backArrowButton
                                }
                            />
                            <p
                                className={ResetWalletPageStyles.subtitle}
                                data-testid="subtitle-reset-wallet"
                            >
                                {t('subTitle')}
                            </p>
                        </div>
                    </div>

                    <div className={ResetWalletPageStyles.mainContent}>
                        <div className={ResetWalletPageStyles.instructionBox}>
                            <InfoIcon
                                testId="icon-reset-instruction"
                                className="mt-[0.1rem]"
                            />
                            <InstructionContent
                                testId="text-reset-instruction"
                                className={
                                    ResetWalletPageStyles.instructionContent
                                }
                                instructionItems={instructionItems}
                            />
                        </div>
                        <SolidButton
                            fullWidth={true}
                            testId="btn-forget-passcode"
                            onClick={handleForgotPasscode}
                            title={t('forgetPasscode')}
                            className={
                                ResetWalletPageStyles.forgetPasscodeButton
                            }
                        />
                    </div>
                </div>
            </div>
        </div>
    );
};
