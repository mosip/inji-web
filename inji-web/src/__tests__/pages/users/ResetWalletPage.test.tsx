import React from 'react';
import {render, screen, fireEvent, waitFor} from '@testing-library/react';
import '@testing-library/jest-dom';
import {useUser} from '../../../hooks/useUser';
import {useCookies} from 'react-cookie';
import {useNavigate, useLocation} from 'react-router-dom';
import {toast} from 'react-toastify';
import {ResetWalletPage} from '../../../pages/users/ResetWalletPage';

jest.mock('react-i18next', () => {
    const translations: {[key: string]: string} = {
        'title': 'Reset Your Wallet',
        'subTitle': 'Reset wallet to regain access and re-download credentials',
        'resetInstruction.question': 'Forgot your passcode?',
        'resetInstruction.info1':
            "If you can't remember your passcode, you can reset it by clicking the button below.",
        'resetInstruction.info2.message':
            "<strong>Please note:</strong> This will remove all your saved cards from the account — such as your <strong>National ID card, Tax ID card, Insurance certificate</strong>, and any other documents you've added.",
        'resetInstruction.info2.highlighter1': 'Please note',
        'resetInstruction.info2.highlighter2':
            'National ID card, Tax ID card, Insurance certificate',
        'resetInstruction.info3.message':
            "After resetting, you'll need to set a new passcode and <strong>re-download your cards</strong> into your Wallet.",
        'resetInstruction.info3.highlighter': 're-download your cards',
        'forgetPasscode': 'Forget Passcode',
        'resetFailure':
            'Something went wrong while resetting your wallet. Please try again in a moment.'
    };

    return {
        useTranslation: () => ({
            t: (key: string, options?: any) => {
                let translatedText = translations[key] || key;

                if (typeof translatedText === 'string' && options) {
                    for (const optKey in options) {
                        if (
                            Object.prototype.hasOwnProperty.call(
                                options,
                                optKey
                            )
                        ) {
                            translatedText = translatedText.replace(
                                new RegExp(`{{${optKey}}}`, 'g'),
                                options[optKey]
                            );
                        }
                    }
                }
                return translatedText;
            },
            i18n: {
                language: 'en',
                changeLanguage: jest.fn()
            }
        }),
        Trans: ({
            children,
            i18nKey,
            values
        }: {
            children?: React.ReactNode;
            i18nKey?: string;
            values?: Record<string, string>;
        }) => {
            let content = translations[i18nKey as string] || i18nKey;

            if (typeof content === 'string' && values) {
                for (const valKey in values) {
                    if (Object.prototype.hasOwnProperty.call(values, valKey)) {
                        content = content.replace(
                            new RegExp(`{{${valKey}}}`, 'g'),
                            values[valKey]
                        );
                    }
                }
            }

            if (typeof content === 'string') {
                return <div dangerouslySetInnerHTML={{__html: content}} />;
            }
            return <>{children}</>;
        }
    };
});

jest.mock('../../../hooks/useUser', () => ({
    useUser: jest.fn()
}));

jest.mock('react-cookie', () => ({
    useCookies: jest.fn()
}));

jest.mock('react-router-dom', () => ({
    useNavigate: jest.fn(),
    useLocation: jest.fn()
}));

jest.mock('react-toastify', () => ({
    toast: {
        error: jest.fn()
    }
}));

global.fetch = jest.fn();

describe('ResetWalletPage Component', () => {
    const mockNavigate = jest.fn();
    const mockRemoveWallet = jest.fn();

    beforeEach(() => {
        jest.clearAllMocks();

        (global.fetch as jest.Mock).mockResolvedValue({
            ok: true,
            json: async () => ({})
        });

        (useUser as jest.Mock).mockReturnValue({
            removeWallet: mockRemoveWallet,
            walletId: 'mock-wallet-id'
        });
        (useCookies as jest.Mock).mockReturnValue([
            {'XSRF-TOKEN': 'mock-xsrf-token'}
        ]);
        (useNavigate as jest.Mock).mockReturnValue(mockNavigate);
        (useLocation as jest.Mock).mockReturnValue({
            state: {walletId: 'location-wallet-id'}
        });
    });

    test('should match snapshot', () => {
        const {asFragment} = render(<ResetWalletPage />);

        expect(asFragment()).toMatchSnapshot();
    });

    test('should render all elements with correct test ids', () => {
        render(<ResetWalletPage />);

        expect(screen.getByTestId('backdrop-reset-wallet')).toBeInTheDocument();
        expect(screen.getByTestId('logo-inji-web')).toBeInTheDocument();
        expect(screen.getByTestId('title-reset-wallet')).toBeInTheDocument();
        expect(screen.getByTestId('btn-back-reset')).toBeInTheDocument();
        expect(
            screen.getByTestId('title-reset-wallet-description')
        ).toBeInTheDocument();
        expect(
            screen.getByTestId('icon-reset-instruction')
        ).toBeInTheDocument();
        expect(
            screen.getByTestId('text-reset-instruction')
        ).toBeInTheDocument();
        expect(screen.getByTestId('text-reset-question')).toBeInTheDocument();
        expect(screen.getByTestId('text-reset-info1')).toBeInTheDocument();
        expect(screen.getByTestId('text-reset-info2')).toBeInTheDocument();
        expect(screen.getByTestId('text-reset-info3')).toBeInTheDocument();
        expect(screen.getByTestId('btn-forget-passcode')).toBeInTheDocument();
    });

    test('should display translated text content correctly', () => {
        render(<ResetWalletPage />);

        expect(screen.getByTestId('title-reset-wallet')).toHaveTextContent(
            'Reset Your Wallet'
        );
        expect(
            screen.getByTestId('title-reset-wallet-description')
        ).toHaveTextContent(
            'Reset wallet to regain access and re-download credentials'
        );
        expect(screen.getByTestId('text-reset-question')).toHaveTextContent(
            'Forgot your passcode?'
        );
        expect(screen.getByTestId('btn-forget-passcode')).toHaveTextContent(
            'Forget Passcode'
        );
        expect(screen.getByTestId('text-reset-info2')).toHaveTextContent(
            "Please note: This will remove all your saved cards from the account — such as your National ID card, Tax ID card, Insurance certificate, and any other documents you've added."
        );
        expect(screen.getByTestId('text-reset-info3')).toHaveTextContent(
            "After resetting, you'll need to set a new passcode and re-download your cards into your Wallet."
        );
    });

    test('should navigate back to /pin when back arrow button is clicked', () => {
        render(<ResetWalletPage />);

        fireEvent.click(screen.getByTestId('btn-back-reset'));

        expect(mockNavigate).toHaveBeenCalledTimes(1);
        expect(mockNavigate).toHaveBeenCalledWith('/pin');
    });

    test('should handle successful wallet reset: delete wallet and navigate to /pin', async () => {
        render(<ResetWalletPage />);

        fireEvent.click(screen.getByTestId('btn-forget-passcode'));

        await waitFor(() => {
            expect(global.fetch).toHaveBeenCalledTimes(1);
            expect(global.fetch).toHaveBeenCalledWith(
                expect.stringContaining('wallets/location-wallet-id'),
                expect.objectContaining({
                    method: 'DELETE',
                    headers: expect.objectContaining({
                        'X-XSRF-TOKEN': 'mock-xsrf-token'
                    }),
                    credentials: 'include'
                })
            );
            expect(mockRemoveWallet).toHaveBeenCalledTimes(1);
            expect(mockNavigate).toHaveBeenCalledTimes(1);
            expect(mockNavigate).toHaveBeenCalledWith('/pin');
            expect(toast.error).not.toHaveBeenCalled();
        });
    });

    test('should handle failed wallet reset: display error toast and not remove wallet or navigate', async () => {
        (global.fetch as jest.Mock).mockResolvedValue({
            ok: false,
            json: async () => ({error: 'Wallet deletion failed'})
        });

        render(<ResetWalletPage />);

        fireEvent.click(screen.getByTestId('btn-forget-passcode'));

        await waitFor(() => {
            expect(global.fetch).toHaveBeenCalledTimes(1);
            expect(mockRemoveWallet).not.toHaveBeenCalled();
            expect(mockNavigate).not.toHaveBeenCalled();
            expect(toast.error).toHaveBeenCalledTimes(1);
            expect(toast.error).toHaveBeenCalledWith(
                'Something went wrong while resetting your wallet. Please try again in a moment.'
            );
        });
    });

    test('should use fallback walletId from useUser when location state is not available', async () => {
        (useLocation as jest.Mock).mockReturnValue({state: null});

        render(<ResetWalletPage />);

        fireEvent.click(screen.getByTestId('btn-forget-passcode'));

        await waitFor(() => {
            expect(global.fetch).toHaveBeenCalledTimes(1);
            expect(global.fetch).toHaveBeenCalledWith(
                expect.stringContaining('wallets/mock-wallet-id'),
                expect.any(Object)
            );
            expect(mockRemoveWallet).toHaveBeenCalledTimes(1);
            expect(mockNavigate).toHaveBeenCalledTimes(1);
            expect(mockNavigate).toHaveBeenCalledWith('/pin');
        });
    });

    test('should handle network error during wallet reset: display error toast and not remove wallet or navigate', async () => {
        (global.fetch as jest.Mock).mockRejectedValue(
            new Error('Network error')
        );

        render(<ResetWalletPage />);

        fireEvent.click(screen.getByTestId('btn-forget-passcode'));

        await waitFor(() => {
            expect(global.fetch).toHaveBeenCalledTimes(1);
            expect(mockRemoveWallet).not.toHaveBeenCalled();
            expect(mockNavigate).not.toHaveBeenCalled();
            expect(toast.error).toHaveBeenCalledTimes(1);
            expect(toast.error).toHaveBeenCalledWith(
                'Something went wrong while resetting your wallet. Please try again in a moment.'
            );
        });
    });
});
