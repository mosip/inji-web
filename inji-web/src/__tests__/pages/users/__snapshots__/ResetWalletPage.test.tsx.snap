// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetWalletPage Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="overflow-hidden fixed inset-0 backdrop-blur-sm bg-black bg-opacity-40 flex flex-col items-center justify-center z-50"
    data-testid="backdrop-reset-wallet"
  >
    <div
      class="overflow-hidden rounded-2xl bg-white flex flex-col items-center justify-start relative w-[90%] sm:w-[85%] md:w-[75%] h-[75%] sm:h-[70%] overflow-y-auto overflow-x-auto shadow-iw-pinPageContainer h-full"
    >
      <div
        class="overflow-hidden absolute inset-0 z-0 flex items-center justify-center pointer-events-none"
        data-testid="background-decorator"
      >
        <div
          class="absolute top-[155px]"
          style="top: 155px;"
        >
          <div
            class="absolute rounded-full border overflow-hidden"
            data-testid="background-decorator-circle-0"
            style="width: 96px; height: 96px; border-width: 1px; border-color: rgba(228, 231, 236, 0.8); top: calc(50% - 48px); left: calc(50% - 48px);"
          />
          <div
            class="absolute rounded-full border overflow-hidden"
            data-testid="background-decorator-circle-1"
            style="width: 192px; height: 192px; border-width: 1px; border-color: rgba(228, 231, 236, 0.7000000000000001); top: calc(50% - 96px); left: calc(50% - 96px);"
          />
          <div
            class="absolute rounded-full border overflow-hidden"
            data-testid="background-decorator-circle-2"
            style="width: 288px; height: 288px; border-width: 1px; border-color: rgba(228, 231, 236, 0.6000000000000001); top: calc(50% - 144px); left: calc(50% - 144px);"
          />
          <div
            class="absolute rounded-full border overflow-hidden"
            data-testid="background-decorator-circle-3"
            style="width: 384px; height: 384px; border-width: 1px; border-color: rgba(228, 231, 236, 0.5); top: calc(50% - 192px); left: calc(50% - 192px);"
          />
          <div
            class="absolute rounded-full border overflow-hidden"
            data-testid="background-decorator-circle-4"
            style="width: 480px; height: 480px; border-width: 1px; border-color: rgba(228, 231, 236, 0.4); top: calc(50% - 240px); left: calc(50% - 240px);"
          />
          <div
            class="absolute rounded-full border overflow-hidden"
            data-testid="background-decorator-circle-5"
            style="width: 576px; height: 576px; border-width: 1px; border-color: rgba(228, 231, 236, 0.30000000000000004); top: calc(50% - 288px); left: calc(50% - 288px);"
          />
          <div
            class="flex items-center justify-center"
            data-testid="logo-inji-web-container"
          >
            <img
              alt="Inji Web Logo"
              data-testid="logo-inji-web"
              src="Logomark.png"
            />
          </div>
        </div>
      </div>
      <div
        class="top-[215px] pb-4 relative flex flex-col items-center justify-center"
      >
        <div
          class="text-center items-center justify-center relative z-20 bg-transparent space-y-4 w-[90%] sm-md:w-[85%] md:w-[50%]"
        >
          <h1
            class="text-xl sm:text-2xl md:text-3xl font-semibold text-iw-darkGreen p-4"
            data-testid="title-reset-wallet"
          >
            Reset Your Wallet
          </h1>
          <div
            class="flex items-start ml-4"
          >
            <button
              class="cursor-pointer"
              data-testid="btn-back-arrow-container"
              type="button"
            >
              <img
                alt="Back Arrow"
                class=""
                data-testid="icon-back-arrow"
                src="BackArrowIcon.svg"
              />
            </button>
            <p
              class="text-iw-textTertiary text-sm sm:text-base font-medium pl-6"
              data-testid="subtitle-reset-wallet"
            >
              Reset wallet to regain access and re-download credentials
            </p>
          </div>
        </div>
        <div
          class="flex flex-col bg-white rounded-xl shadow-iw-pinPageContainer mt-8 items-center relative z-20 px-4 sm:px-6 md:px-10 py-3 sm:py-5 md:py-7 space-y-4 w-[90%] sm-md:w-[85%] md:w-[50%]"
        >
          <div
            class="bg-iw-paleGray rounded-xl flex items-start justify-center gap-2 p-4 pr-10"
          >
            <img
              alt="Info"
              class="mt-[0.1rem]"
              data-testid="icon-reset-instruction"
              src="InfoIcon.svg"
            />
            <div
              class="flex flex-col items-left gap-2 sm:gap-4 rounded-lg"
              data-testid="text-reset-instruction"
            >
              <p
                class="text-xs sm:text-sm md:text-md text-left font-montserrat font-semibold text-black"
                data-testid="text-reset-question"
              >
                Forgot your passcode?
              </p>
              <p
                class="text-xs sm:text-sm md:text-md text-left font-montserrat font-normal text-black"
                data-testid="text-reset-info1"
              >
                If you can't remember your passcode, you can reset it by clicking the button below.
              </p>
              <p
                class="text-xs sm:text-sm md:text-md text-left font-montserrat font-normal text-black"
                data-testid="text-reset-info2"
              />
              <div>
                <strong>
                  Please note:
                </strong>
                 This will remove all your saved cards from the account â€” such as your 
                <strong>
                  National ID card, Tax ID card, Insurance certificate
                </strong>
                , and any other documents you've added.
              </div>
              <p />
              <p
                class="text-xs sm:text-sm md:text-md text-left font-montserrat font-normal text-black"
                data-testid="text-reset-info3"
              />
              <div>
                After resetting, you'll need to set a new passcode and 
                <strong>
                  re-download your cards
                </strong>
                 into your Wallet.
              </div>
              <p />
            </div>
          </div>
          <div
            class="w-full"
          >
            <div
              class="w-full bg-iw-primary text-center cursor-pointer items-center text-iw-text font-bold text-sm px-6 py-3 rounded-lg shadow hover:shadow-lg mr-1  bg-gradient-to-r from-iw-primary to-iw-secondary mt-3 mb-4 sm:mt-3 sm:mb-4"
            >
              <button
                class="text-white font-bold rounded-lg"
                data-testid="btn-forget-passcode"
              >
                Forget Passcode
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
